server/lib/tuist/qa/simulator_controller.ex

```diff
@@ -0,0 +1,121 @@
+defmodule Tuist.QA.SimulatorController do
+  @moduledoc """
+  Controller for interacting with iOS Simulator via xcrun simctl.
+  """
+
+  alias Tuist.QA.SimulatorDevice
+
+  @doc """
+  Lists all available simulator devices.
+
+  Options:
+    - runtime_identifier: Filter devices by runtime identifier (e.g. "com.apple.CoreSimulator.SimRuntime.iOS-18-4")
+    - name: Filter devices by name (e.g. "iPhone 16")
+
+  Returns an array of `SimulatorDevice`.
+  """
+  def devices(opts \\ []) do
+    runtime_identifier = Keyword.get(opts, :runtime_identifier)
+    name = Keyword.get(opts, :name)
+
+    with {output, 0} <- System.cmd("xcrun", ["simctl", "list", "devices", "--json"]),
+         {:ok, %{"devices" => devices_by_runtime}} <- Jason.decode(output) do
```

Comment:
Should we use Elixir's built-in [JSON](https://hexdocs.pm/elixir/JSON.html) module?